### PR TITLE
classfields: eval early errors

### PR DESCRIPTION
**Pull request by spectranaut**
@leo take a look whenever :)

### CODE REVIEW

**Comment by leo** [comment]
@spectranaut Can you folks please start using the right usernames 😄

**Comment by spectranaut** [comment]
oooppss @leobalter XD

File: src/class-fields/initializer-eval-super/cls-expr-fields-indirect-eval.template [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+---*/
+
+A = class {}
+
+var executed = false;
+C = class extends A {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/*{ execution error }*/, function() {
```
Comment (leobalter) [inline src/class-fields/initializer-eval-super/cls-expr-fields-indirect-eval.template]: interesting, without generating the files I can't tell if the script handles multiple words

**Comment by rwaldron** [comment]
> Can you folks please start using the right usernames

😂

File: src/class-fields/initializer-eval/cls-expr-fields-indirect-eval.template [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+/*---
+esid: sec-performeval-rules-in-initializer
+path: language/expressions/class/fields-indirect-
+name: indirect eval
+---*/
+
+var executed = false;
+C = class {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
```
Comment (rwaldron) [inline src/class-fields/initializer-eval/cls-expr-fields-indirect-eval.template]: Missing a closing `)`

File: src/class-fields/initializer-eval/cls-expr-fields-eval.template [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/*---
+esid: sec-performeval-rules-in-initializer
+path: language/expressions/class/fields-direct-
+name: direct eval
+---*/
+
+var executed = false;
+C = class {
+  x = eval('executed = true; /*{ initializer }*/;';
```
Comment (rwaldron) [inline src/class-fields/initializer-eval/cls-expr-fields-eval.template]: Missing a closing `)`

File: src/class-fields/initializer-eval/cls-decl-fields-indirect-eval.template [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+/*---
+esid: sec-performeval-rules-in-initializer
+path: language/statements/class/fields-indirect-
+name: indirect eval
+---*/
+
+var executed = false;
+class C = {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
```
Comment (rwaldron) [inline src/class-fields/initializer-eval/cls-decl-fields-indirect-eval.template]: Missing a closing `)`


...Ok, I'm not going to be annoying and mark all of these—please fix each file :)

File: src/class-fields/eval-err-contains-arguments.case [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2017 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: `arguments` in StatementList of eval
```
Comment (leobalter) [inline src/class-fields/eval-err-contains-arguments.case]: the initial ` triggers a failure on the yaml parser. This applies to the other cases as well.

File: src/class-fields/eval-err-contains-superproperty.case [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  These static semantics are applied by PerformEval when a direct eval call occurs outside of a MethodDefinition.
+  ScriptBody:StatementList
+
+    It is a Syntax Error if StatementList Contains SuperProperty.
+features: [class-fields]
+template: initializer-eval-super
+---*/
+
+//- initializer
+super().x
```
Comment (leobalter) [inline src/class-fields/eval-err-contains-superproperty.case]: I believe this should be `super.x` to match a SuperProperty:

```
SuperProperty:
  super [ Expression ]
  super . IdentifierName
```

And this case may also be duplicated with `super['x']`

File: src/class-fields/initializer-eval-super/cls-decl-fields-indirect-eval.template [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+---*/
+
+class A = {}
+
+var executed = false;
+class C extends A = {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/*{ execution error }*/, function() {
```
Comment (leobalter) [inline src/class-fields/initializer-eval-super/cls-decl-fields-indirect-eval.template]: the reutilization of the templates using execution and early errors fields is great! +1

**Comment by spectranaut** [comment]
ok addressed all comments and added generated files, should be good to go @rwaldron @leobalter 

(btw, leo, the space in the template keywords definitely did not work XD removed them)

File: src/class-fields/initializer-eval-super/cls-decl-fields-eval.template [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+---*/
+
+class A = {}
+
+var executed = false;
+class C extends A = {
+  x = eval('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/* early error */, function() {
```
Comment (leobalter) [inline src/class-fields/initializer-eval-super/cls-decl-fields-eval.template]: oops

File: src/class-fields/initializer-eval-super/cls-decl-fields-indirect-eval.template (hunk 7 above)
Comment (leobalter) [reply src/class-fields/initializer-eval-super/cls-decl-fields-indirect-eval.template]: the templates still remain with the "spaced" names.

File: src/class-fields/initializer-eval/cls-decl-fields-eval.template [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+path: language/statements/class/fields-direct-
+name: direct eval
+---*/
+
+var executed = false;
+class C = {
+  x = eval('executed = true; /*{ initializer }*/;');
+}
+
+assert.throws(/* earlyerror */, function() {
```
Comment (leobalter) [inline src/class-fields/initializer-eval/cls-decl-fields-eval.template]: not all of them, thou :)

Comment (spectranaut) [reply src/class-fields/initializer-eval/cls-decl-fields-eval.template]: ohhhnoooo... what's happening?

Comment (spectranaut) [reply src/class-fields/initializer-eval/cls-decl-fields-eval.template]: oh oops I see, forgot "{ }" XD

**Comment by spectranaut** [comment]
ok @leo fixed those last few mistakes!

**Comment by leo** [comment]
GOD 😄 username

**Pull request merged**